### DISCUSSION TITLE
T-BEAM V1.2 with AXP2101, battery charging.

### DISCUSSION THREAD

**Discussion by AndRZJt1** [dl9rdz/rdz_ttgo_sonde#432] [TTGO firmware]
Good morning,
I have T-BEAM V1.2 with AXP2101
I have a problem with displaying battery voltage and charging the battery.
After connecting the charger, the charging LED does not light up.
I have the following software installed: master_v0.9.4-B17-full

![T-Beam v1 2](https://github.com/dl9rdz/rdz_ttgo_sonde/assets/165378175/e2a020b8-e5d3-4950-93a5-fbded8761327)

Please let me know if this problem can be solved.

Thank you and best regards

**Comment by alx6ths** [comment]
Hello, I have the same board and iam having the same or similar problem. The Battery status page does not show any information, no voltage, no temperature, or anything else.
Is there any solution for this please?

![20240420_154511](https://github.com/dl9rdz/rdz_ttgo_sonde/assets/8413517/9e22589f-8f9e-454b-966c-02feabab57fd)

Thank you, Libor

**Comment by hr-ru** [comment]
Yes, may I ask you, is WiFi working on your T-Beam with the binary firmware image?

I have one T-Beam 1.2 now for testing, but I have the issue that it does not connect to an WiFi network (but WiFi works in AP mode). Changing the bootloader to 40 Mhz flash SPI (instead of 80 Mhz) eliminates the problem, but I do not understand why. 

Also my travis CI credit for building the images has expired. devel20240419 is available as source (and should show voltage on the T-Beam 1.2) but not yet tested, and not yet built as ready to use image.

**Reply by alx6ths** [reply]
Iam using the master_v0.9.4-B17 at this moment, should i try the devel instead?
The Wifi works without problem at this moment.

**Reply by hr-ru** [reply]
devel: Yes, but devel20240419 is not yet on the download page, only source on github

**Reply by alx6ths** [reply]
aha, i see. Not sure how i can use it, iam not a big friend of Platformio :-D

**Reply by AndRZJt1** [reply]
@alx6ths
I have the same battery problem.  The LED does not light up after connecting the charger.

**Reply by alx6ths** [reply]
I have finaly managed to upload the version devel20240419 to my t-beam, but i have no changes on the Battery status screen. Only one thing i noticed is the missing "-" symbol in front of the 0.00V value. No real voltage, no Temp, no USB: whatever information should be there.
Am i still missing something?

Thank you.

**Reply by dl9rdz** [reply]
I missed that the function should return mV, not V. 
USB voltage, Battery voltage, and Battery status (present-B / charging-C / not present-U) should work now (devel 20240514)

For those with the tiny LCD display, there is now also a new Scanner screen (8) that shows Battery status, Battery voltage and USB voltage (replace 0 with 8 in the config)

**Reply by alx6ths** [reply]
Works now, thank you very very much Hansi !!
Although, only the voltage is displayed, but not the charging and discharging current for example. Is this also supported?

Thank you,
Libor, OK1ALX

**Reply by dl9rdz** [reply]
No, the AXP2101 does not have the possibility to measure the current

**Reply by alx6ths** [reply]
ok, thank you for the clarification.

**Reply by AndRZJt1** [reply]
@alx6ths can you send the compiled version (devel 20240514), I can't cope with the Arduino IDE configuration. 
Thank you

**Reply by alx6ths** [reply]
@AndRZJt1 Hi, I'd love to help you with this, but I'm in a similar situation with Visual Studio Code and unfortunately I don't have such a file. The firmware file created during compilation is about 4 times smaller and cannot be used separately with the Flash Tool. I compiled it and uploaded it directly in Visual Studio Code. A friend helped me with it and I'm glad that it somehow worked out in the end.

Maybe someone can advise how to create a working bin file...

**Reply by dl9rdz** [reply]
Its now available on the download page as binary image (and as update)

**Reply by AndRZJt1** [reply]
Thank you very much for a great job and a job well done.

**Comment by AdamM68** [comment]
I also have LILYGO T-BEAM V1.2 AXP2101,  rdzTTGOsonde version dev20250218
everything works OK but !
- on battery page I see : temp -1       ... is this option working/compatibile with this model ?
- software is recognizing board as: Autodetect info: Fingerprint 55 (TTGO T-Beam (V0.7), 0.9" OLED@21,22)
.. but my board is V1.2
can I change something in setup to have my board correctly recognized ?
